### PR TITLE
Fix lgtm alerts

### PR DESCRIPTION
**Pull request by xiemaisi**
This fixes a few minor issues found by [lgtm](https://lgtm.com); for more details see https://lgtm.com/projects/g/adobe/brackets/alerts/.

~~The fixes in the first three commits are fairly obvious, the last two commits involve some educated guessing as to the intended semantics of the code involved.~~

### CODE REVIEW

File: src/extensibility/Package.js
Comment (petetnt) [inline src/extensibility/Package.js]: This makes sense but I am not 100% sure either, @swmitra?

**Review by petetnt — COMMENTED** [review]
Cool stuff @xiemaisi, LGTM but that last one is bit tricky so someone else should check it too.

**Comment by xiemaisi** [comment]
I've removed the last two commits with the less obvious fixes; I'll put them up as a separate PR.

**Comment by xiemaisi** [comment]
@petent, does this look safer now?

**Comment by xiemaisi** [comment]
@petetnt, does this look safer now?

File: src/extensions/default/JavaScriptCodeHints/HintUtils.js
Comment (petetnt) [inline src/extensions/default/JavaScriptCodeHints/HintUtils.js]: This looks bit scary to me, but didn't manage to find any differences after running some tests cases by hand 🤔

**Review by petetnt — APPROVED** [review]
LGTM with a comment, someone else should do a double check too. @saurabh95 could you do that 🙏

File: src/extensions/default/JavaScriptCodeHints/HintUtils.js
Comment (saurabh95) [inline src/extensions/default/JavaScriptCodeHints/HintUtils.js]: This one looks scary to me as well. Will do some testing around it.

Comment (xiemaisi) [inline src/extensions/default/JavaScriptCodeHints/HintUtils.js]: It's not as scary as you might think: the original code looks for a substring of `t.value` that contains zero or more backslashes, followed by something that isn't a backslash, followed by a double quote. This is the same as simply looking for a substring consisting of something that isn't a backslash followed by a double quote, which is what the new code does.

(Note that lgtm actually only highlighted the fact that `[\\\\]` is redundant and means the same as `[\\]` or indeed `\\`.)

Comment (saurabh95) [inline src/extensions/default/JavaScriptCodeHints/HintUtils.js]: Oh ok. seems good to me. We can merge this after 1.10 release

Comment (petetnt) [inline src/extensions/default/JavaScriptCodeHints/HintUtils.js]: Good stuff @xiemaisi, merging this in and will land in 1.11. Hopefully we get the other fixes found by `lgtm` in too.

**Pull request merged**
